### PR TITLE
Vmo 3702 jsonschemas for each block types

### PR DESCRIPTION
**Pull request by safydy-r**
[script output overview]
```
$ yarn jsonschema:generate-blocks
yarn run v1.22.10
$ node scripts/jsonschema-generate-blocks.js
Generating schema for IAdvancedSelectOneBlock
Generating schema for ICaseBlock
Generating schema for ILogBlock
Generating schema for IMessageBlock
Generating schema for INumericResponseBlock
Generating schema for IOpenResponseBlock
Generating schema for IOutputBlock
Generating schema for IPrintBlock
Generating schema for IReadBlock
Generating schema for IRunFlowBlock
Generating schema for ISelectOneResponseBlock
Generating schema for ISetGroupMembershipBlock
Done in 36.57s.
```
Once the PR is approved, I will remove all generated files. They were put here just for PR approval.

### CODE REVIEW

**Comment by bzabos** [comment]
As per discussion w/ @safydy, I think we can try removing references to properties on other interfaces in favour of explicitly defining the type; eg: replace this:
```
interface IAdvancedSelectOneBlockConfig {
  x: IResource['uuid']
}
```

… with this:
```
interface IAdvancedSelectOneBlockConfig {
  x: string // IResource['uuid']
}
```

**Comment by safydy** [comment]
@bzabos , this is ready for review

File: package.json
Comment (safydy-r) [inline package.json:10]: @bzabos , I'd propose to change the output file name for IContainer, `flowSpecJsonSchema.json` came from the target branch.
`flowSpecJsonSchema.json` could be confusing as a developer might think this is for `IFlow` interface, while it's for `IContainer`. How about `floipSpecJsonSchema.json` or directly `IContainer.json` ?

File: dist/resources/IMessageBlock.json
Comment (safydy-r) [inline dist/resources/IMessageBlock.json:49]: @bzabos , Json schema recognises empty string as valid (even we put it under `required`). Any idea how we can change our flow-runner interface to automatically generate constraints? eg:
- `non-empty` string for `name`s`
- positive number for `IFlow.interaction_timeout`

**Comment by safydy-r** [comment]
@bzabos , and I'm realising we don't have Interfaces for `Photo` & `Location Response` Blocks. Any idea ?

File: dist/resources/IMessageBlock.json
Comment (safydy-r) [reply dist/resources/IMessageBlock.json:49]: Answered here https://votomobile.slack.com/archives/CGGFKUP4Y/p1618852982013000

File: src/flow-spec/IFlow.ts
Comment (safydy-r) [inline src/flow-spec/IFlow.ts]: FYI @markboots , it seems like the `@TJS-` annotations are not working with `ts-json-schema-generator`, but it is working without the prefix. I don't know why, and I don't see a doc explaining this behaviour so far.  After investigating & seeing this issue  https://github.com/vega/ts-json-schema-generator/issues/445, I just tried `@pattern` instead of `@TJS-pattern`

Comment (safydy-r) [reply src/flow-spec/IFlow.ts]: cc: @bzabos & @poojahpatil90

File: src/flow-spec/IBlockExit.ts
Comment (safydy-r) [inline src/flow-spec/IBlockExit.ts:34]: I removed this as it's not matching with current implementation so far, we're using simple `string` (eg: `Default`, `Error`, etc) rather than `uuid` here.
@bzabos , shall we update the spec description to reflect this usage or shall we change the way we set this `label` in flow-builder ?

File: dist/resources/IMessageBlock.json
Comment (safydy-r) [reply dist/resources/IMessageBlock.json:49]: this is resolved, I implemented json schema annotations

**Comment by safydy-r** [comment]
**Note**: 
According to @bzabos , we will merge this & the Mark's PR (vmo-2573) once we have approval on Validation tickets PRs

File: src/domain/validation/FlowContainerValidator.ts
Comment (safydy-r) [inline src/domain/validation/FlowContainerValidator.ts:18]: FYI: @bzabos 
This was needed to correctly use json schema format. Solution from here: https://github.com/ajv-validator/ajv/issues/1295#issuecomment-754679968

**Review by bzabos — APPROVED** [review]
Thanks @safydy, this looks good!

**Pull request merged**
